pkg/target_determinator.go

```diff
@@ -221,6 +236,16 @@ func LoadIncompleteMetadata(context *Context, rev LabelledGitRev, targets Target
 	if err != nil {
 		return queryInfo, cleanupFunc, fmt.Errorf("failed to query at %s in %v: %w", rev, context.WorkspacePath, err)
 	}
+
+	if context.CompareQueriesAroundAnalysisCacheClear {
+		if !reflect.DeepEqual(queryInfoBeforeClear.MatchingTargets, queryInfo.MatchingTargets) {
```

Comment:
Can we implement an `Equal(other QueryResults)` method on `QueryResults` to encapsulate the use of `reflect.DeepEqual`?